moto/ec2/responses/vpcs.py

```diff
@@ -251,9 +256,19 @@ def describe_vpc_endpoints(self) -> str:
         vpc_end_points = self.ec2_backend.describe_vpc_endpoints(
             vpc_end_point_ids=vpc_end_points_ids, filters=filters
         )
+
+        security_group_ids = []
+        for service in vpc_end_points:
+            security_group_ids.extend(service.security_group_ids)
```

Comment:
The failing error is because `service.security_group_ids` can be None - so you're effectively running `security_group_ids.extend(None)`, which throws the `TypeError: 'NoneType' object is not iterable` that you can see in the  logs.